.github/workflows/trunk-tagging.yml

```diff
@@ -0,0 +1,224 @@
+name: trunk-tagging
```

Comment:
Have you considered wrapping the logic in this workflow into a GHA and find a home for it outside of PyTorch?  The input to that GHA can be the trio `repo_name`, `commit_sha`, and `workflow_names` with more than one workflow.  The benefit is that other repos could use it too.  This could come later if there is a need, for now, we can keep this in mind.